rts/Lua/LuaMatrixImpl.h

```diff
@@ -0,0 +1,164 @@
+	bool UnitMatrix(const unsigned int unitID, const sol::optional<bool> mult, sol::this_state L);
+	bool UnitPieceMatrix(const unsigned int unitID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L);
+
+	bool FeatureMatrix(const unsigned int featureID, const sol::optional<bool> mult, sol::this_state L);
+	bool FeaturePieceMatrix(const unsigned int featureID, const unsigned int pieceNum, const sol::optional<bool> mult, sol::this_state L);
+
+	bool ProjectileMatrix(const unsigned int projID, const sol::optional<bool> mult, sol::this_state L);
+
+	void ScreenViewMatrix(const sol::optional<bool> mult);
+	void ScreenProjMatrix(const sol::optional<bool> mult);
+	void ScreenViewProjMatrix(const sol::optional<bool> mult);
+
+	void SunViewMatrix(const sol::optional<bool> mult);
+	void SunProjMatrix(const sol::optional<bool> mult);
+	void SunViewProjMatrix(const sol::optional<bool> mult);
+
+	void Ortho(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
+	void Frustum(const float left, const float right, const float bottom, const float top, const float near, const float far, const sol::optional<bool> mult);
+
+	void CameraViewMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraProjMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewProjMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraProjInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraViewProjInverseMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+	void CameraBillboardMatrix(const sol::optional<unsigned int> cameraIdOpt, const sol::optional<bool> mult);
+
+	tuple16f GetAsScalar()
+	{
+		return std::make_tuple(
+			mat[0], mat[1], mat[2], mat[3],
+			mat[4], mat[5], mat[6], mat[7],
+			mat[8], mat[9], mat[10], mat[11],
+			mat[12], mat[13], mat[14], mat[15]
+		);
+	}
+
+	sol::as_table_t<std::array<float, 16>> GetAsTable()
+	{
+		return sol::as_table(std::array<float, 16> {
+			mat[0], mat[1], mat[2], mat[3],
+			mat[4], mat[5], mat[6], mat[7],
+			mat[8], mat[9], mat[10], mat[11],
+			mat[12], mat[13], mat[14], mat[15]
+		});
+	}
+
+public:
+	const CMatrix44f& GetMatRef() const { return  mat; }
+	const CMatrix44f* GetMatPtr() const { return &mat; }
+public:
+	sol::as_table_t<std::array<float, 4>> operator* (const sol::table& tbl) const {
+		float4 f4 {0.0f, 0.0f, 0.0f, 1.0f};
+		for (auto i = 1; i <= 4; ++i) {
+			const sol::optional<float> maybeValue = tbl[i];
+			if (!maybeValue)
+				continue;
+
+			f4[i - 1] = maybeValue.value();
```

Comment:
might as well write this as 
`f4[i - 1] = tbl[i].value_or(f4[i - 1]);`
saves you an if-check and a few lines (am I nitpicking?)